### ISSUE TITLE
[1.12.2] JourneyMap "save map" attempts to write invalid filename on Win 10

### ISSUE THREAD

**Issue by ThePlasmaRailgun** [TeamJM/journeymap#99]
#### Instructions for Submitting an Issue
1. Ask for help on Discord (https://discord.gg/eP8gE69) #support if you're unsure this is a new issue.
2. Please search for existing issues before submitting a new one.
3. Please add the correct MC<version> label to your issue. Example: "[1.12.2] Waypoints are upside down"
4. Completely fill out the template below. Failure to do so is likely to get the issue closed immediately.
5. Prove you read these instructions by deleting this section before submitting. ;)
---
  
#### JourneyMap Version:

1.12.1-5.5.2b1

#### Forge Version:

1.12.2-14.23.4.2760

#### Describe the problem in repeatable steps:

1. Auto map a world
2. Press "save map" button
3. Confirm the saved .png name contains pipe characters `|`
4. Access screenshots directory and confirm .png does not exist because `|` character is not allowed in Windows filenames

#### Relevant errors from `.minecraft/journeymap/journeymap.log`:
```[20:50:02] [main/INFO] [ChatLog] JourneyMap: Saving map to screenshots: 2019-04-05_20.50.01_world-mc-heightmap_overworld_0|day|_.png
[20:57:18] [JM-task-6/ERROR] [MapSaver] ar.com.hjg.pngj.PngjInputException: Could not open for writeC:\Users\Fin\AppData\Roaming\RedstoneBees\screenshots\2019-04-05_20.57.16_world-mc-heightmap_overworld_0|night|_.png
	at ar.com.hjg.pngj.PngHelperInternal2.ostreamFromFile(PngHelperInternal2.java:30)
	at ar.com.hjg.pngj.PngHelperInternal.ostreamFromFile(PngHelperInternal.java:284)
	at ar.com.hjg.pngj.PngWriter.<init>(PngWriter.java:71)
	at journeymap.client.io.PngjHelper.mergeFiles(PngjHelper.java:41)
	at journeymap.client.io.MapSaver.saveMap(MapSaver.java:83)
	at journeymap.client.task.multi.SaveMapTask.performTask(SaveMapTask.java:42)
	at journeymap.client.thread.RunnableTask$Inner.run(RunnableTask.java:81)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.FileNotFoundException: C:\Users\Fin\AppData\Roaming\RedstoneBees\screenshots\2019-04-05_20.57.16_world-mc-heightmap_overworld_0|night|_.png (The filename, directory name, or volume label syntax is incorrect)
	at java.io.FileOutputStream.open0(Native Method)
	at java.io.FileOutputStream.open(FileOutputStream.java:270)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
	at ar.com.hjg.pngj.PngHelperInternal2.ostreamFromFile(PngHelperInternal2.java:28)
```

[Relevant MSDN article on disallowed characters and reserved filenames](https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file#naming-conventions)

**Comment by mysticdrew** [comment]
Why are you using an old beta? I don't know if that issue has been fixed since then, maybe it has. You shouldn't run beta mods for long.

**Comment by mysticdrew** [comment]
This is currently not an issue the development version. As per discord, I suggested to update to 5.5.4 since the he is on an old beta. Maybe working correctly in 5.5.4, for sure it is fixed in 5.5.5 release.
